engine/apps/api/serializers/custom_button.py

```diff
@@ -51,15 +52,31 @@ def validate_data(self, data):
             return None
 
         try:
-            json.loads(data)
-        except ValueError:
-            raise serializers.ValidationError("Data has incorrect format")
-
-        try:
-            Template(data)
+            template = Template(data)
         except TemplateError:
             raise serializers.ValidationError("Data has incorrect template")
 
+        try:
+            rendered = template.render(
+                {
+                    # Validate that the template can be rendered with a JSON-ish alert payload.
+                    # We don't know what the actual payload will be, so we use a defaultdict
+                    # so that attribute access within a template will never fail
+                    # (provided it's only one level deep - we won't accept templates that attempt
+                    # to do nested attribute access).
+                    # Every attribute access should return a string to ensure that users are
+                    # correctly using `tojson` or wrapping fields in strings.
+                    # If we instead used a `defaultdict(dict)` or `defaultdict(lambda: 1)` we
+                    # would accidentally accept templates such as `{"name": {{ alert_payload.name }}}`
+                    # which would then fail at the true render time due to the
+                    "alert_payload": defaultdict(lambda: ""),
```

Comment:
Just curious if there's some particular reason to use `defaultdict(lambda: "")` instead of `defaultdict(str)` here?

Reply:
Oh, nope, I think I had a `lambda: somethingelse` halfway through and just replaced the `somethingelse` but yours is much nicer. Thanks! Fixed in eabc402.